build.gradle.kts

```diff
@@ -0,0 +1,190 @@
+	}
+
+	// Dependency Management for Subprojects
+	dependencies {
+		constraints {
+			// Annotations
+			api(group = "org.jetbrains", name = "annotations", version = "20.1.0")
+
+			// Caching
+			api(group = "com.github.ben-manes.caffeine", name = "caffeine", version = "2.8.6")
+
+			// Apache Commons
+			api(group = "commons-configuration", name = "commons-configuration", version = "1.10")
+
+			// Rate Limiting
+			api(group = "com.github.vladimir-bukhtoyarov", name = "bucket4j-core", version = "4.7.0")
+
+			// Event Dispatcher
+			api(group = "com.squareup.okhttp3", name = "okhttp", version = "4.9.0")
+			api(group = "com.github.philippheuer.events4j", name = "events4j-core", version = "0.9.5")
+			api(group = "com.github.philippheuer.events4j", name = "events4j-handler-simple", version = "0.9.5")
+			api(group = "com.github.philippheuer.events4j", name = "events4j-handler-spring", version = "0.9.5")
+			api(group = "com.github.philippheuer.events4j", name = "events4j-handler-reactor", version = "0.9.5")
+
+			// Credential Manager
+			api(group = "com.github.philippheuer.credentialmanager", name = "credentialmanager", version = "0.1.1")
+
+			// HTTP Client
+			api(group = "io.github.openfeign", name = "feign-slf4j", version = "11.0")
+			api(group = "io.github.openfeign", name = "feign-okhttp", version = "11.0")
+			api(group = "io.github.openfeign", name = "feign-jackson", version = "11.0")
+			api(group = "io.github.openfeign", name = "feign-hystrix", version = "11.0")
+
+			// WebSocket
+			api(group = "com.neovisionaries", name = "nv-websocket-client", version = "2.10")
+
+			// Hystrix
+			api(group = "com.netflix.hystrix", name = "hystrix-core", version = "1.5.18")
+		}
+		// Apache Commons
+		api(group = "commons-io", name = "commons-io", version = "2.8.0")
+		api(group = "org.apache.commons", name = "commons-lang3", version = "3.11")
+
+		// Logging
+		api(group = "org.slf4j", name = "slf4j-api", version = "1.7.30")
+
+		// Jackson BOM
+		implementation(platform("com.fasterxml.jackson:jackson-bom:2.12.1"))
+
+		// Test
+		testImplementation(platform("org.junit:junit-bom:5.7.0"))
+		testImplementation(group = "org.junit.jupiter", name = "junit-jupiter")
+		testImplementation(group = "ch.qos.logback", name = "logback-classic", version = "1.2.3")
+	}
+
+	publishing {
+		publications {
+			create<MavenPublication>("main") {
+				from(components["java"])
+				artifactId = project.artifactId
+				pom.default()
+			}
+		}
+	}
+
+	// Source encoding
+	tasks {
+		withType<JavaCompile> {
+			options.encoding = "UTF-8"
+		}
+
+		withType<Jar> {
+			archiveBaseName.set(artifactId)
+		}
+
+		val delombok by getting(io.freefair.gradle.plugins.lombok.tasks.Delombok::class)
+		val jar by getting(Jar::class)
+		val relocateShadowJar by creating(com.github.jengelman.gradle.plugins.shadow.tasks.ConfigureShadowRelocation::class) {
+			target = shadowJar.get()
+			prefix = "com.github.twitch4j.shaded.${"$version".replace(".", "_")}"
+		}
```

Comment:
I have use this task. [Reference](https://github.com/johnrengelman/shadow/issues/629)